crates/ol/mempool/src/state.rs

```diff
@@ -0,0 +1,1933 @@
+            self.ordering_index.insert(priority, tx_data.txid);
+
+            // Add to per-account index
+            let account_id = entry.tx.target();
+            self.by_account
+                .entry(account_id)
+                .or_default()
+                .insert(tx_data.txid);
+
+            self.entries.insert(tx_data.txid, entry);
+        }
+
+        // Set next_insertion_id to continue from max + 1
+        self.next_insertion_id = max_insertion_id.saturating_add(1);
+
+        // Recompute stats
+        self.stats.mempool_size = self.entries.len();
+        self.stats.total_bytes = self.entries.values().map(|e| e.size_bytes).sum();
+
+        Ok(())
+    }
+}
+
+/// Combined state for the service (context + mutable state).
+#[derive(Debug)]
+pub(crate) struct MempoolServiceState {
+    ctx: Arc<MempoolContext>,
+    state: MempoolState,
+}
+
+impl MempoolServiceState {
+    /// Create new mempool service state.
+    #[expect(dead_code, reason = "will be used via builder")]
+    pub(crate) fn new(
+        config: OLMempoolConfig,
+        storage: Arc<NodeStorage>,
+        current_tip: OLBlockCommitment,
+    ) -> Self {
+        let ctx = Arc::new(MempoolContext::new(config, storage));
+        Self::new_with_context(ctx, current_tip)
+    }
+
+    /// Create new mempool service state with an existing context.
+    /// Used for testing.
+    pub(crate) fn new_with_context(
+        ctx: Arc<MempoolContext>,
+        current_tip: OLBlockCommitment,
+    ) -> Self {
+        Self {
+            ctx: ctx.clone(),
+            state: MempoolState::new(current_tip),
+        }
+    }
+
+    /// Load existing transactions from database.
+    pub(crate) fn load_from_db(&mut self) -> OLMempoolResult<()> {
+        self.state.load_from_db(&self.ctx)
+    }
+
+    /// Set the current tip (called when chain progresses).
+    #[cfg_attr(not(test), expect(dead_code, reason = "will be used via builder"))]
+    pub(crate) fn set_current_tip(&mut self, tip: OLBlockCommitment) {
+        self.state.set_current_tip(tip);
+    }
+
+    /// Get the current tip.
+    #[expect(dead_code, reason = "will be used via builder")]
+    pub(crate) fn current_tip(&self) -> &OLBlockCommitment {
+        self.state.current_tip()
+    }
+
+    /// Handle submit transaction command.
+    pub(crate) async fn handle_submit_transaction(
+        &mut self,
+        tx_bytes: Vec<u8>,
+    ) -> OLMempoolResult<OLTxId> {
+        // Deserialize transaction from SSZ bytes
+        let tx = OLMempoolTransaction::from_ssz_bytes(&tx_bytes)
+            .map_err(|e| OLMempoolError::Serialization(format!("SSZ decode error: {:?}", e)))?;
+
+        // Compute transaction ID
+        let txid = tx.compute_txid();
+
+        // Add to mempool
+        self.state.add_transaction(&self.ctx, tx).await?;
+
+        Ok(txid)
+    }
+
+    /// Handle best transactions command (returns all transactions in priority order).
+    pub(crate) async fn handle_best_transactions(
```

Comment:
as noted [above](https://github.com/alpenlabs/alpen/pull/1185/files#r2644283282), am I wrong that there's no handling here -- this is just a get function.